socket/controlmessage.go

```diff
@@ -0,0 +1,118 @@
+package socket
```

Comment:
this is all copied verbatim from .... ? Might need a comments in these file. Do we need keep in sync with these files? (we basically vendoring by copying the code?)